src/components/documentManager.ts

```diff
@@ -0,0 +1,532 @@
+                    item.ImageTags.Primary
+            );
+        else return null;
+    }
+
+    // Another tool for showItem
+    private static getLogoUrl(item: BaseItemDto): string | null {
+        if (item.ImageTags?.Logo)
+            return JellyfinApi.createUrl(
+                'Items/' + item.Id + '/Images/Logo/0?tag=' + item.ImageTags.Logo
+            );
+        else if (item.ParentLogoItemId && item.ParentLogoImageTag)
+            return JellyfinApi.createUrl(
+                'Items/' +
+                    item.ParentLogoItemId +
+                    '/Images/Logo/0?tag=' +
+                    item.ParentLogoImageTag
+            );
+        else return null;
+    }
+
+    // showItem takes an item and shows it on a details page
+    public static showItem(item: BaseItemDto): void {
+        // stop cycling backdrops
+        this.clearBackdropInterval();
+
+        this.setAppStatus('details');
+        this.setWaitingBackdrop(item);
+
+        this.setLogo(this.getLogoUrl(item) || '');
+        this.setOverview(item.Overview || '');
+        this.setGenres(item?.Genres?.join(' / ') ?? null);
+        this.setDisplayName(item);
+
+        const detailRating = document.getElementById('detailRating');
+        this.setMiscInfo(item);
+
+        if (detailRating) {
+            detailRating.innerHTML = this.getRatingHtml(item);
+        }
+
+        const playedIndicator = document.getElementById('playedIndicator');
+
+        if (playedIndicator) {
+            if (item?.UserData?.Played) {
+                playedIndicator.style.display = 'block';
+                playedIndicator.innerHTML =
+                    '<span class="glyphicon glyphicon-ok"></span>';
+            } else if (item?.UserData?.UnplayedItemCount) {
+                playedIndicator.style.display = 'block';
+                playedIndicator.innerHTML = item.UserData.UnplayedItemCount.toString();
+            } else {
+                playedIndicator.style.display = 'none';
+            }
+        }
+
+        let detailImageUrl = this.getPrimaryImageUrl(item);
+
+        if (
+            item?.UserData?.PlayedPercentage &&
+            item?.UserData?.PlayedPercentage < 100 &&
+            !item.IsFolder
+        ) {
+            this.setHasPlayedPercentage(false);
+            this.setPlayedPercentage(item.UserData.PlayedPercentage);
+
+            if (detailImageUrl != null)
+                detailImageUrl +=
+                    '&PercentPlayed=' +
+                    item.UserData.PlayedPercentage.toString();
+        } else {
+            this.setHasPlayedPercentage(false);
+            this.setPlayedPercentage(0);
+        }
+
+        this.setDetailImage(detailImageUrl);
+    }
+
+    // It has some overloadings
+    public static showItemId(itemId: string): void {
+        const url = JellyfinApi.createUserUrl('Items/' + itemId);
+
+        JellyfinApi.authAjax(url, 'GET').then(function (item: BaseItemDto) {
```

Comment:
might not be a big deal in this case but its probably worthwhile returning this promise so whatever calls this function can wait for it to finish and be sure that the dom changes have been applied